Add option for 'NamedList' component to group items

Merseleo opened the pull request:
Closes #394 

## What I have made

<!-- write down what changes have been made, what was removed/added/changed -->
<!-- e.g. I added a new button, which does x -->

- I added two options to the `NamedList` component that allow setting a grouping function and group labels
   - If set, the items in the named list are grouped using the `groupBy` method and a group header (followed by a separator) is added before each group

## Checklist

Either tick or cross out the items that do not apply (using \~\~example text\~\~) and give a reason why the item does not apply.

- [x] I have updated the documentation accordingly and commented my code
- [x] I have added tests that prove my fix is effective or that my feature works
  - ~~[ ] unit tests~~
  - [x] integration tests
  - ~~[ ] end-to-end tests~~
- [x] (for reviewer) I have checked the implementation against the requirements

Commit:
feat: add option for 'NamedList' component to group items

When the items should be grouped, as on the home page the open reviews, the group name and a separator are inserted before each group items

Commit:
test: add test case for grouping items in 'NamedList' component

domienderle commented on src/lib/components/composites/list/NamedList.svelte:
```diff
@@ -97,11 +139,21 @@ This can be e.g. a search bar.
+                    {#each Object.entries(groupBy(loadedItems, groupSelector)) as [key, values] (key)}
+                        <div class="space-y-1">
+                            {#await groupLabels}
+                                <h3 class="italic">Loading</h3>
```
```suggestion
                                <h1 class="italic">Loading...</h1>
```
What do you think about this?

Merseleo replied:
I thought we agreed on not using "..." when loading something? But if you want this for clarity I can add them.

domienderle replied:
That is true. Since the loading is italic it should be enough then.

domienderle commented on src/lib/components/composites/list/NamedList.svelte:
```diff
@@ -97,11 +139,21 @@ This can be e.g. a search bar.
-                {/each}
+                {#if groupSelector !== undefined}
+                    {#each Object.entries(groupBy(loadedItems, groupSelector)) as [key, values] (key)}
+                        <div class="space-y-1">
```
```suggestion
                        <div class="space-y-1 mt-10">
```
What do you think about this?

Merseleo replied:
Good idea, but in my opinion 40px is bit too much. What do you think of the new solution?

domienderle replied:
This is also fine. I guess that's a matter of taste in the end. But looks good as it is now.

domienderle commented on src/lib/components/composites/list/NamedList.svelte:
```diff
@@ -97,11 +139,21 @@ This can be e.g. a search bar.
+                            {#await groupLabels}
+                                <h3 class="italic">Loading</h3>
+                            {:then loadedGroupLabels}
+                                <h3>{loadedGroupLabels?.[key] ?? "Unknown"}</h3>
```
```suggestion
                                <h1>{loadedGroupLabels?.[key] ?? "Unknown"}</h1>
```

Merseleo replied:
In my opinion the group header should not have a high-level heading the header of the entire list. So semantically `h1` is wrong imo. Additionally, it seems too big in my opinion. But if you want me to increase the size of the group headers and can change the text size style separately.

Merseleo replied:
But as I tested some different header sizes, `h2` seems to look the best and is more correct, as the `PaperListEntry` or other components use `h2` as well and this text should not be bigger than the heading as well. Are you ok with that?

domienderle replied:
Yeah, that's a good point. Since it looks good as it is now, I think we are good here.

Commit:
refactor: slightly redesign the 'NamedList' component when the items should be grouped

Before each group header, except for the first one, a gap is added; the group header are h2 and not h3 heading anymore

Merseleo reviewed (commented):
Thank you for the feedback. I played around with some different designs and I think based on your feedback I found a good solution. What do you think?

domienderle reviewed (approved):
Yes I am fine with that. Looks good now. Thank you.
